command/command.ts

```diff
@@ -2695,6 +2708,13 @@ type TypedOption<
   : F extends `-${infer Name}` ? BooleanOption<Name, CO, IsRequired<R, D>, D>
   : Record<string, unknown>;
 
+type TypedArgumentDefinition<
```

Comment:
I was solving for `<arg1> <arg2> <others...>` -> `[ "foo", "bar", [ "one", "two", "three"] ]`.

Since `TypedArguments` works recursively, a variadic argument at the end of an arg list would _always_ be treated as a "JustRestArg", and therefore appended to a flat args array, which would have had a result type of `[ "foo", "bar", "one", "two", "three" ]` which doesn't seem as helpful.